src/main/commandServer/__tests__/httpCommandServer.spec.ts

```diff
@@ -0,0 +1,96 @@
+   * 2. Rancher Desktop isn't running.
+   * There are edge cases where this test might fail, such as when Rancher Desktop is starting up
+   * so the VM exists but the command server hasn't started yet. For the purposes of running this in CI,
+   * or by developers during a typical edit-test-fix cycle, these are edge cases we can ignore for now.
+   */
+  notRunningWinIt("should fail to run rdctl shell when server isn't running", async() => {
+    try {
+      await spawnFile(rdctlPath, ['list-settings'], { stdio: 'pipe' });
+      console.log('Skipping rdctl shell failure test because the rdctl server is running.');
+    } catch (err: any) {
+      const stderr = err.stderr ?? '';
+
+      if (stderr.match(/Error.*\/v\d\/settings.*dial tcp.*connect: connection refused/)) {
+        try {
+          const { stdout } = await spawnFile(rdctlPath, ['shell', 'echo', 'abc'], { stdio: 'pipe' });
+
+          expect(stdout).toEqual('Running rdctl shell should have failed.');
+        } catch (err: any) {
+          const stderr = err.stderr ?? '';
+
+          expect(stderr).toContain("Either run 'rdctl start' or start the Rancher Desktop application first");
+          expect(stderr).toMatch(/(?:The Rancher Desktop VM needs to be created)|(?:The Rancher Desktop VM needs to be in state "Running" in order to execute 'rdctl shell', but it is currently in state)/);
+        }
+      } else {
+        expect(stderr).toMatch(/Error.*\/v\d\/settings.*dial tcp.*connect: connection refused/);
+      }
+    }
+  });
+
+  runningWinIt("should fail to run on Windows when there's no r-d WSL", async() => {
+    try {
+      const { stdout, stderr } = await spawnFile('wsl', ['-l', '-v'], { stdio: 'pipe', encoding: 'utf16le' });
+      const splitLines = stdout.split(/\r?\n/);
+      const lines = splitLines.filter(line => (line ?? '').match(/rancher-desktop\s/));
+
+      expect(stderr).toEqual('');
+      if (lines[0]?.match(/Running/)) {
+        console.log(`Skipping test because `);
+
+        return;
+      }
+
+      if (stdout.match(/.*rancher-desktop\s+Running.*/)) {
+        console.log('WSL rancher-desktop is available.');
+
+        return;
+      }
+      try {
+        const { stdout } = await spawnFile(rdctlPath, ['shell', 'echo', 'abc'], { stdio: 'pipe' });
+
+        expect(stdout).toEqual('Running rdctl shell should have failed.');
+      } catch (err: any) {
+        const stdout = err.stdout ?? '';
+        const stderr = err.stderr ?? '';
+
+        expect(stdout).toBe('');
+        expect(stderr).toMatch(/(?:The Rancher Desktop WSL needs to be running in order to execute 'rdctl shell', but it currently is not.)|(?:The Rancher Desktop WSL needs to be in state "Running" in order to execute 'rdctl shell', but it is currently in state)/);
+        expect(stderr).toContain("Either run 'rdctl start' or start the Rancher Desktop application first");
+      }
+    } catch (err: any) {
+      console.log(`wslconfig failed: ${ err }`);
```

Comment:
`wslconfig`?

Reply:
I originally ran `wslconfig /l` before I remembered the `wsl -lv` code.